lib/govuk_index/publishing_event_worker.rb

```diff
@@ -56,17 +48,48 @@ def process_action(actions, routing_key, payload)
       else
         logger.info("#{routing_key} -> SKIPPED #{presenter.base_path} #{presenter.type}")
       end
+
+    # Rescuing as we don't want to retry this class of error
+    rescue MissingBasePath => e
+      return if DOCUMENT_TYPES_WITHOUT_BASE_PATH.include?(payload["document_type"])
+      GovukError.notify(e, extra: { message_body: payload })
+        # Unpublishing messages for something that does not exist may have been
+        # processed out of order so we don't want to notify errbit but just allow
+        # the process to continue
+    rescue NotFoundError
+      logger.info("#{payload['base_path']} could not be found.")
+      Services.statsd_client.increment('govuk_index.not-found-error')
+    rescue UnknownDocumentTypeError
+      logger.info("#{payload['document_type']} document type is not known.")
+      Services.statsd_client.increment('govuk_index.unknown-document-type')
     end
 
-    def process_response(response)
-      # we are only expecting to process a single message at a time
-      if response['items'].count > 1
-        Services.statsd_client.increment('govuk_index.elasticsearch.multiple_responses')
-        raise MultipleMessagesInElasticsearchResponse
+    def process_responses(responses, messages)
```

Comment:
I think the name of this method should still be singular, since we're only sending one request to ES.

Reply:
changed, but the variable names may still need some thought.